doc/dds-core-xtypes/README.md

```diff
@@ -120,94 +122,93 @@ The type consistency, could be a set of the following *QoS polities*:
+    int value = data[2].value<int>(); // get value from position 2 of the collection.
+    data[2] = dynamic_data_representing_a_value;
+    WritableDynamicDataRef ref = data[2]; //references to a DynamicData that represents a collection
+    ```
 1. `DynamicData` represents a `StringType`
-The same as `CollectionType` plus:
-  ```c++
-  data.value<std::string>("Hello data!"); //sets the string value
-  data.string("Hello again!"); // shortcut version for string
-  const std::string& s1 = data.value<std::string>(); //read the string value
-  const std::string& s2 = data.string(); // shortcut version for string
-  ```
+    Same as `CollectionType` plus:
+    ```c++
+    data.value<std::string>("Hello data!"); //sets the string value
+    data.string("Hello again!"); // shortcut version for string
+    const std::string& s1 = data.value<std::string>(); //read the string value
+    const std::string& s2 = data.string(); // shortcut version for string
+    ```
 1. `DynamicData` represents a `WStringType`
-The same as `CollectionType` plus:
-  ```c++
-  data.value<std::wstring>(L"Hello data! \u263A"); //sets the string value
-  data.wstring(L"Hello again! \u263A"); // shortcut version for string
-  const std::wstring& s1 = data.value<std::wstring>(); //read the string value
-  const std::wstring& s2 = data.wstring(); // shortcut version for string
-  ```
+    Same as `CollectionType` plus:
+    ```c++
+    data.value<std::wstring>(L"Hello data! \u263A"); //sets the string value
+    data.wstring(L"Hello again! \u263A"); // shortcut version for string
+    const std::wstring& s1 = data.value<std::wstring>(); //read the string value
+    const std::wstring& s2 = data.wstring(); // shortcut version for string
+    ```
 1. `DynamicData` represents a `SequenceType`
-The same as `CollectionType` plus:
-  ```c++
-  data.push(42); // push back new value to the sequence.
-  data.push(dynamic_data_representing_a_value);
-  ```
+    Same as `CollectionType` plus:
+    ```c++
+    data.push(42); // push back new value to the sequence.
+    data.push(dynamic_data_representing_a_value);
+    ```
 
 #### References to `DynamicData`
 There are two ways to obtain a reference to a `DynamicData`:
 1. `ReadableDynamicDataRef` for constant references.
 2. `WritableDynamicDataRef` for mutable references.
 
-A reference no contain any value, only points to an already existing `DynamicData`, or part of it.
-You can get a reference when access with the `[]` operator or calling the `ref()` and `cref()` methods of the `DynamicData`.
-You will obtain a `ReadableDynamicDataRef` or a `WritableDynamicDataRef` depending if theses methods are
-calling from a `const DynamicData` or from a `DynamicData` itself.
+A reference does not contain any value and only points to an already existing `DynamicData`, or to part of it.
+You can obtain a reference by accessing data with `[]` operator or by calling `ref()` and `cref()`, such methods being provided by `DynamicData`.
```

Comment:
```suggestion
You can obtain a reference by accessing data with `[]` operator or by calling `ref()` and `cref()`.
```
I would remove the last part because the next line explain it (more or less), and these methods are also provided by the read/write references. Can be confused.